### PR TITLE
Fix progressive loading of KTX skybox textures

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7ffc54c58fe6081515cc416469d6aaf3810b92d1 available at: https://gist.github.com/86233e7443efdcc49b47edd34074d443<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-7ffc54c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-7ffc54c.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: fa44071d72098ac7c6e234bd2235e1b6f25eb741 available at: https://gist.github.com/86233e7443efdcc49b47edd34074d443<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-fa44071.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-fa44071.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a886963e20699192cfa0effccb312a0840f1fc97 available at: https://gist.github.com/86233e7443efdcc49b47edd34074d443<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-a886963.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-a886963.dmg

**Comment by AndrewMeadows** [comment]
Jenkins: rebuild this please.

**Comment by hifi-gustavo** [comment]
Build console output for commit: a886963e20699192cfa0effccb312a0840f1fc97 available at: https://gist.github.com/86233e7443efdcc49b47edd34074d443<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-a886963.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-a886963.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 87a685c4f9302ba15bc9b02237eacb2d2982fe68 available at: https://gist.github.com/86233e7443efdcc49b47edd34074d443<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-87a685c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-87a685c.dmg

**Comment by huffman** [comment]
To test:

 1. Set a zone skybox on your domain to https://hifi-public.s3.amazonaws.com/huffman/ktx/SKY-Forest-2-toon2.ktx
 2. Close interface, clear your cache
 3. Open interface. You should see the skybox load a low resolution relatively fast (it shouldn't be black for a long period of time) and slowly improve in quality as it loads fully.

**Comment by claudioshg** [comment]
testing

**Comment by claudioshg** [comment]
### Desktop PC - Windows 10 Home - CPU 4.0GHz Intel i7 - GTX1070 - RAM 16GB - Tested in Desktop Mode

Tested: Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10364-87a685c.exe

PASS:
- I set a zone skybox to https://hifi-public.s3.amazonaws.com/huffman/ktx/SKY-Forest-2-toon2.ktx
- I then closed the interface and deleted the PR10364 folder at LOCAL.
- I reopened the interface and saw the skybox load the low res background instantly. The 'black' blinked in and out of existence. The skybox's low res image then improved in quality as seconds passed.
- I re-tested 3 times and every time the zone skybox loaded as intended.

File: libraries/networking/src/ResourceCache.cpp [hunk 1]
```diff
@@ -533,13 +533,13 @@ void Resource::ensureLoading() {
...
 
 void Resource::setLoadPriority(const QPointer<QObject>& owner, float priority) {
-    if (!(_failedToLoad || _loaded)) {
+    if (!(_failedToLoad)) {
         _loadPriorities.insert(owner, priority);
     }
 }
 
 void Resource::setLoadPriorities(const QHash<QPointer<QObject>, float>& priorities) {
-    if (_failedToLoad || _loaded) {
```
Comment (Atlante45) [inline libraries/networking/src/ResourceCache.cpp]: I know you didn't add that, but why would those 2 be different? `!_failedToLoad` and `_failedToLoad`

**Pull request merged**